pipeline/main.py

```diff
@@ -1,279 +0,0 @@
-    labels = graph.ndata.pop('label').to(device).long()
-
-    # Extract node features
-    feats = graph.ndata.pop('feat').to(device)
-    n_features = feats.shape[-1]
-
-    # retrieve masks for train/validation/test
-    train_mask = graph.ndata.pop('train_mask')
-    val_mask = graph.ndata.pop('val_mask')
-    test_mask = graph.ndata.pop('test_mask')
-
-    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze().to(device)
-    val_idx = torch.nonzero(val_mask, as_tuple=False).squeeze().to(device)
-    test_idx = torch.nonzero(test_mask, as_tuple=False).squeeze().to(device)
-
-    graph.to(device)
-
-    # Step 2: Create model =================================================================== #
-    mixhopgcn_model = MixHopGCN(in_dim=n_features,
-                                hid_dim=args.hid_dim,
-                                out_dim=n_classes,
-                                num_layers=args.num_layers,
-                                p=args.p,
-                                input_dropout=args.input_dropout,
-                                layer_dropout=args.layer_dropout,
-                                activation=torch.tanh,
-                                batchnorm=True)
-    
-    mixhopgcn_model = mixhopgcn_model.to(device)
-
-    # Step 3: Create training components ===================================================== #
-    loss_fn = nn.CrossEntropyLoss()
-    opt = optim.SGD(mixhopgcn_model.parameters(), lr=args.lr, weight_decay=args.lamb)
-    scheduler = optim.lr_scheduler.StepLR(opt, args.step_size, gamma=args.gamma)
-
-    # Step 4: training epoches =============================================================== #
-    acc = 0
-    no_improvement = 0
-    epochs = trange(args.epochs, desc='Accuracy & Loss')
-    
-    for _ in epochs:
-
-        # Training and validation using a full graph
-        mixhopgcn_model.train()
-
-        logits = mixhopgcn_model.forward(graph, feats)
-
-        # compute loss
-        train_loss = loss_fn(logits[train_idx], labels[train_idx])
-        train_acc = torch.sum(logits[train_idx].argmax(dim=1) == labels[train_idx]).item() / len(train_idx)
-
-        valid_loss = loss_fn(logits[val_idx], labels[val_idx])
-        valid_acc = torch.sum(logits[val_idx].argmax(dim=1) == labels[val_idx]).item() / len(val_idx)
-
-        # backward
-        opt.zero_grad()
-        train_loss.backward()
-        opt.step()
-
-        # Print out performance
-        epochs.set_description('Train Acc {:.4f} | Train Loss {:.4f} | Val Acc {:.4f} | Val loss {:.4f}'.format(
-            train_acc, train_loss.item(), valid_acc, valid_loss.item()))
-        
-        if valid_acc < acc:
-            no_improvement += 1
-            if no_improvement == args.early_stopping:
-                print('Early stop.')
-                break
-        else:
-            no_improvement = 0
-            acc = valid_acc
-        
-        scheduler.step()
-
-    # Test with mini batch after all epoch
```

Comment:
I don't think you need this "# Test with mini batch after all epoch"

Reply:
Removed.